### DISCUSSION TITLE
Informal meetup at CppCon 2023 @ 12.30 Monday 2 Oct

### DISCUSSION THREAD

**Discussion by bluetarpmedia** [hsutter/cppfront#650] [Suggestions]
EDIT - Now confirmed at **12.30 Monday 2 Oct**

Is anyone going to CppCon this year? If so, we could arrange an informal meetup one lunchtime/evening to chat about Cpp2.
* Show off a pull request you're working on and get feedback
* Chat in person about some of the Cppfront GH discussion topics

**Comment by gregmarr** [comment]
I will be there.

**Comment by hsutter** [comment]
Sounds like that would be a good [Open Content submission](https://cppcon.org/open-content-submissions/) -- would someone like to fill out the form to suggest it? Note that Monday and Tuesday evenings are taken up with panels I have to attend, but I should be able to attend on Wednesday or Thursday lunchtime or evening if an Open Content session proposal for this is accepted and scheduled then.

FYI, @filipsajdak will also be giving a a cppfront-related talk at CppCon, currently scheduled for Thursday after lunch: [Advancing cppfront with Modern C++: Refining the Implementation of is, as, and UFCS](https://cppcon2023.sched.com/event/1Qtfz/advancing-cppfront-with-modern-c-refining-the-implementation-of-is-as-and-ufcs)

**Reply by filipsajdak** [reply]
Yes, I will be there!

**Reply by JohelEGP** [reply]
I'm looking forward to your talk, too, @filipsajdak.
(Offline after it's published to YouTube, as I don't attend conferences).

**Comment by hsutter** [comment]
As a fallback, if an Open Content submission isn't made and accepted, we can probably ask the organizers for a meeting room where we can meet ad-hoc. But an OC session would be better because IIUC it would already have AV support in the room for showing things on screen and such, and it would be easier to advertise (including on the Sched.com schedule ) so we know when/where to go (and so that other people can know about it in case anyone not on this discussion thread is interested).

**Reply by filipsajdak** [reply]
Who should send the submission? Is it better when you do it or should one of us send it?

**Reply by hsutter** [reply]
Yes someone else please go ahead and send the submission and work on the OC arrangements, and just let me know what arrangements end up being made. I'll be happy to show up, and should be able to as long as it's not Mon/Tue evening. Thanks for the idea!

**Reply by gregmarr** [reply]
Wednesday and Thursday evening open content may also overlap with the Diversity and Speakers' dinners, for those that are attending those.  I am checking on the times for the dinners now.

Update: we do expect them to overlap, so we should avoid at least the Thursday Speakers' dinner.

**Reply by hsutter** [reply]
Ah, I knew about the dinners but I thought they started at around 18:00 and the open content didn't start until 20:30 ... ?

**Reply by gregmarr** [reply]
OC is 18:30-20:00.  I believe it's Lighting Talks in the 20:30 slot on Wednesday and Thursday, same slot as the panels on Monday and Tuesday.

**Reply by bluetarpmedia** [reply]
Maybe a lunchtime OC session would be better?

**Reply by gregmarr** [reply]
I didn't see this in the list of proposed Open Content sessions when I looked a day or two ago.  (It's not published yet, I'm the conference registrar.)  Has anyone submitted?

**Reply by hsutter** [reply]
Seeing your note, I just went ahead and put in a submission... thanks for the reminder.

**Comment by JohelEGP** [comment]
Can't this be pinned somehow?

**Reply by hsutter** [reply]
Pinned!

**Comment by bluetarpmedia** [comment]
CppCon has scheduled the Open Content session at 12.30pm on Monday 2 Oct.

>This session is an informal “BoF + show-and-tell” meetup for people interested in Cpp2/cppfront.
Here are some ideas:
>- Do you have a pull request you're working on? Show it off and get feedback
>- Chat in person about some of the GitHub [cppfront Discussions](https://github.com/hsutter/cppfront/discussions) topics
>So this would be a BoF session
>- Other related topics welcome!

**Comment by hsutter** [comment]
Just to get a bit organized, if you see this in time 😁 and are considering talking about a subject...

What's your name and subject you're thinking of talking about at the meetup? Note: We should have a projector and screen if you want to show off code or demonstrate something!

I have two suggestions:

(1) I've asked Filip to talk about his PR for `is` and `as`

(2) I want to make a short bullet list of things I have planned and have some discussion/feedback on what you suggest my priorities should be... the short list is:
- self-hosting (writing more of cppfront in Cpp2 to gain experience)
- writing a Cpp2 user guide (I put this on the list because I planned to get to that "later" but I keep seeing more feedback that people want a guide soon)
- bug fixes in Cpp1 code generation (and generally catching up on issues and PRs)
- additional Cpp2 features (e.g., exception handling, extending contracts to class invariants, starting to implement some support for Lifetime static analysis, ...)
- writing a metafunction that generates an output file at compile time, which can be source code in another language (e.g., exposing a C++ type to C# or Java by generating a native C# or Java type that wraps it)
- ...

What else would anyone like to chat about or present?

**Reply by bluetarpmedia** [reply]
@hsutter If there's a spare 5 mins or so then I'd love to present some ideas about Cpp2 & safety. (Neil Henderson)

**Comment by hsutter** [comment]
Thanks to everyone who came to the BoF today!

Thanks to @filipsajdak for presenting on `is`/`as` implementation improvements using concepts and subsumption rules to eliminate many of the negative `requires` clause conditions the implementation is currently using.

Thanks to @bluetarpmedia for a discussion on directions for bounds/lifetime safety, for the latter with both static analysis and also dynamic analysis injection possibilities.

I also presented the above five general areas I'm planning to do work on, and polled the room on "if you could pick just two of those five, which would they be?" and the results were:

- Clear winner: Writing a Cpp2 user's guide. That's something I'd planned for later, but I hear the feedback that I should prioritize it sooner.

- Tied for second place: Adding more Cpp2 features (e.g., EH, class invariants, Lifetime static analysis) and writing a metafunction that generates compile-time file output (e.g., to generate code in another language, such as a Java/Swift wrapper for a C++ object by writing `my_interface: @java_interface type = { ... }`.

- About 5-6 hands each: Self-hosting (writing more of cppfront in Cpp2, which is mainly for my own use and my own experience with writing Cpp2) and catching up on bugs/issues/PRs.

Thanks for the feedback, and for spending Monday lunchtime on this project! It was a great start to the week for me, and I hope you enjoy the rest of CppCon week.

**Reply by gregmarr** [reply]
EH was my specific reason for voting for more features.

**Reply by JohelEGP** [reply]
> About 5-6 hands each:

I'm curious.
What was the quorum?

**Comment by bluetarpmedia** [comment]
> I'm curious. What was the quorum?

I think there was about 30-40 people there.

**Reply by JohelEGP** [reply]
That's a lot!
About how many hands did the first and second place get?

**Reply by bluetarpmedia** [reply]
IIRC most of the room raised hands for the User Guide, and maybe a third of the room each for the tied second place options.

**Reply by JohelEGP** [reply]
Thank you for sharing that.

**Comment by gregmarr** [comment]
@hsutter Is it still useful to have this pinned?

**Reply by hsutter** [reply]
Done - thanks for waiting as I catch up on the backlog
